src/main/java/seedu/address/model/person/Salary.java

```diff
@@ -38,7 +38,7 @@ public Salary() {
      * @return A boolean stating if the string is valid or not.
      */
     public static boolean isValidSalary(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return (test.matches(VALIDATION_REGEX) && test.length() <= 15);
```

Comment:
Nice, but this check can be added to the validation regex:
`"\\d{1,15}"`

Reply:
That's cleaner, will make the changes! Thanks!